cli/src/main/java/com/devonfw/tools/ide/tool/PluginMaps.java

```diff
@@ -0,0 +1,42 @@
+package com.devonfw.tools.ide.tool;
+
+import com.devonfw.tools.ide.tool.ide.PluginDescriptor;
+
+import java.util.Map;
+
+/**
+ * A wrapper class for holding two maps of plugin descriptors: one keyed by plugin ID and the other keyed by plugin name.
+ */
+public class PluginMaps {
+  private final Map<String, PluginDescriptor> mapById;
+  private final Map<String, PluginDescriptor> mapByName;
+
+  /**
+   * Constructs a new PluginMaps object with the specified maps.
+   *
+   * @param mapById   A map with plugin IDs as keys and corresponding plugin descriptors as values.
+   * @param mapByName A map with plugin names as keys and corresponding plugin descriptors as values.
+   */
+  public PluginMaps(Map<String, PluginDescriptor> mapById, Map<String, PluginDescriptor> mapByName) {
+    this.mapById = mapById;
+    this.mapByName = mapByName;
+  }
+
+  /**
+   * Returns the map of plugin descriptors keyed by plugin ID.
+   *
+   * @return The map of plugin descriptors keyed by plugin ID.
+   */
+  public Map<String, PluginDescriptor> getById() {
+    return mapById;
+  }
+
+  /**
+   * Returns the map of plugin descriptors keyed by plugin name.
+   *
+   * @return The map of plugin descriptors keyed by plugin name.
+   */
+  public Map<String, PluginDescriptor> getByName() {
+    return mapByName;
+  }
```

Comment:
I always like if we create dedicated structures but then we should also follow encapsulation:
Create the maps internally and remove them from `PluginBasedCommandlet` and instead delegate registration.

```suggestion
  
  private final IdeContext context;

  /**
   * The constructor.
   * @param context the {@link IdeContext}.
   */
  public PluginMaps(IdeContext context) {
    super();
    this.context = context;
    this.mapById = new HashMap<>();
    this.mapByName = new HashMap<>();
  }

  /**
   * @return the {@link PluginDescriptor} for the given {@link PluginDescriptor#getId() ID} or {@code null} if not found.
   */
  public PluginDescriptor getById(String id) {
    return this.mapById.get(id);
  }

  /**
   * @return the {@link PluginDescriptor} for the given {@link PluginDescriptor#getName() name} or {@code null} if not found.
   */
  public PluginDescriptor getByName(String name) {
    return this.mapByName.get(name);
  }
  
  public Collection<PluginDescriptor> getPlugins() {
    
    Map<String, PluginDescriptor> map = this.mapById;
    if (map.isEmpty()) {
      map = this.mapByName; // potentially plugins for this tool have no ID
    }
    return Collections.unmodifyableCollection(map.values());
  }
  
  /**
   * Registers a {@link PluginDescriptor} to this map.
   */
  protected void add(PluginDescriptor descriptor) {
    put(decriptor.getName(), descriptor, this.mapByName);
    String id = descriptor.getId();
    if ((id != null) && !id.isEmpty()) {
      put(id, descriptor, this.mapById);
    }
  }
  
  private void put(String key, PluginDescriptor descriptor, Map<String, PluginDescriptor> map) {
    PluginDescriptor duplicate = map.put(key, descriptor);
    if (duplicate != null) {
      this.context.info("Plugin with key {} was {} but got overridden by {}", key, duplicate, descriptor);
    }
  }
```